src-java/kilda-persistence-tinkerpop/src/test/java/org/openkilda/persistence/ferma/repositories/FermaHaFlowPathRepositoryTest.java

```diff
@@ -0,0 +1,381 @@
+        assertEquals(meterId, actualEdge.getMeterId());
+    }
+
+    private HaSubFlow createSubFlow(String subFlowId, SwitchId switchId, int port, int vlan, int innerVlan) {
+        HaSubFlow subFlow = HaSubFlow.builder()
+                .subFlowId(subFlowId)
+                .endpointSwitchId(switchId)
+                .endpointPort(port)
+                .endpointVlan(vlan)
+                .endpointInnerVlan(innerVlan)
+                .status(FlowStatus.UP)
+                .build();
+        haSubFlowRepository.add(subFlow);
+        return subFlow;
+    }
+
+    static List<PathSegment> buildSegments(HaFlowPath path, Switch switch1, Switch switch2, Switch switch3) {
+        PathSegment segment1 = PathSegment.builder()
+                .pathId(path.getHaPathId()).srcSwitch(switch1).destSwitch(switch3).build();
+        PathSegment segment2 = PathSegment.builder()
+                .pathId(path.getHaPathId()).srcSwitch(switch3).destSwitch(switch2).build();
+        return Lists.newArrayList(segment1, segment2);
+    }
+
+    private List<PathSegment> buildSegments(HaFlowPath path) {
+        return buildSegments(path, switch1, switch2, switch3);
+    }
+
+    static Set<HaSubFlowEdge> buildSubFlowEdges(HaSubFlow subFlow1, HaSubFlow subFlow2) {
+        HaSubFlowEdge edge1 = buildSubFlowEdge(HA_FLOW_ID_1, subFlow1, METER_ID_3);
+        HaSubFlowEdge edge2 = buildSubFlowEdge(HA_FLOW_ID_1, subFlow2, METER_ID_4);
+        return Sets.newHashSet(edge1, edge2);
+    }
+
+    private Set<HaSubFlowEdge> buildSubFlowEdges() {
+        return buildSubFlowEdges(subFlow1, subFlow2);
+    }
+
+    private HaFlowPath createHaFlowFlowWithSegmentsAndEdges(
+            PathId pathId, long bandwidth, FlowSegmentCookie cookie, String sharedGroupId, MeterId sharedMeterId,
+            MeterId yPointMeterId, Switch sharedSwitch, SwitchId yPointSwitchId, GroupId yPointGroupId) {
+        HaFlowPath path = buildHaFlowPath(
+                pathId, bandwidth, cookie, sharedGroupId, sharedMeterId, yPointMeterId, sharedSwitch,
+                yPointSwitchId, yPointGroupId);
+        haFlowPathRepository.add(path);
+
+        haFlow.addPaths(path);
+        path.setSegments(buildSegments(path));
+        path.setHaSubFlowEdges(buildSubFlowEdges());
+        assertEquals(2, path.getHaFlow().getSubFlows().size());
+        return path;
+    }
+
+    private HaFlowPath createHaFlowPath(
+            PathId pathId, long bandwidth, FlowSegmentCookie cookie, String sharedGroupId, MeterId sharedMeterId,
+            MeterId yPointMeterId, Switch sharedSwitch, SwitchId yPointSwitchId, GroupId yPointGroupId) {
+        HaFlowPath path = buildHaFlowPath(
+                pathId, bandwidth, cookie, sharedGroupId, sharedMeterId, yPointMeterId, sharedSwitch,
+                yPointSwitchId, yPointGroupId);
+        haFlowPathRepository.add(path);
+        return path;
+    }
+
+    static HaFlowPath buildHaFlowPath(
```

Comment:
could be private

Reply:
Maybe it's not a big a deal and just a matter of habits and style. Nevertheless, in my experience, dependencies between tests brings more harm than benefits (btw the same for inheritance in tests). I see it like dependency between impl modules. By its nature, those methods are some helpers or utility methods and they don't belong to the class responsible for testing. Those utility methods have better reusability when they are separated from tests. It could be a static method in a helper class or a service class via composition if there is a need to set up some other objects like other repositories.
I see it in the similar way as we all can write methods here and there like check whether the value is null and then set a default one, or we can extract it to some utility class called ObjectUtils and use it everywhere. Yeah, another point is when somebody wants to extract it, they might find it's already been done before.

Reply:
Good point. I moved builders into separate class